### ISSUE TITLE
Tomboy "Not responding" when logging-out of Mate Desktop

### ISSUE THREAD

**Issue by acrothoi2** [tomboy-notes/tomboy#32]
when logging-out of the Mate Desktop, I get the following message:
![not-responding](https://cloud.githubusercontent.com/assets/24640690/21298291/32bf0986-c55d-11e6-889d-0081b0616250.png)

After clicking "Log Out Anyway", it takes several seconds (~30) before I finally get logged-out.

this issue did not exist in Fedora 21 (final updates) with Mate Desktop 1.10.2 and Tomboy 1.15.4
this issue did not exist in Fedora 22 (initial release) with Mate Desktop 1.10.0 and Tomboy 1.15.4

this issue **exists** in Fedora 22 (initial release) with Tomboy 1.15.4 after running:
dnf update mate*
(list of package updates resulting: http://pastebin.com/cmN907mM)

This issue **still exists** with Tomboy 1.15.6 in Fedora 24 with Mate Desktop 1.16.1

**Comment by alex-ter** [comment]
That Tomboy 1.15.6 on Fedora 24,  is it self-compiled? Do you by chance have config.log from the compilation?

**Comment by acrothoi2** [comment]
i didn't compile it; "thm" compiled the package for the fedora repos.

does this provide the information you need? (build.log, maybe?):
https://koji.fedoraproject.org/koji/buildinfo?buildID=824760

**Comment by alex-ter** [comment]
Oh, I see, yes, that's good enough, thanks

**Comment by j-4** [comment]
Same is true on my Gnome, Fedora 24 setup. When trying to logout, gnome waits for tomboy to quit, which can take forever. Sometimes there is a notification, if you want to kill tomboy. If i close tomboy first (and therefore remove it from the notification area) and then log out, its really fast.

**Comment by alex-ter** [comment]
Thanks folks, I'm going to look into this in the next several days' timeframe. It's very nice to see that the latest version made it into a distro that fast, so I'm keen on helping the momentum

**Comment by alex-ter** [comment]
So the immediate reason for this behavior is that Tomboy, while being a registered DBus client (for the Remote Control functionality), does not implement the `QueryEndSession` DBus signal handler, which apparently was not a requirement by gnome-/mate-session-manager previously.

I'm not that much into DBus yet,so it would take me some time to get through the documents, but I think I at least have all the necessary information available and we know the reason at least 😄

**Comment by alex-ter** [comment]
Correction - actually the necessary callback implementations are actually there, but apparently not being called properly, even though at the first glance they are still ok. Looking further into this, stay tuned.

**Comment by alex-ter** [comment]
At the end of the day it turned out the Tomboy itself is okay - we are implementing necessary signal handlers for quite a while. It's the `dbus-sharp` package in Fedora, which was not updated in ages (version 0.7.0) and has a certain bug, which was fixed later than that and which causes the signals not to be propagated to the application.

Specific commits are:
1. https://github.com/mono/dbus-sharp/commit/608328ba1e0ccc8d0387c244793b229bcc184648
1. https://github.com/mono/dbus-sharp/commit/76f43e6cc25eae25b658831111b37b10376311d2

I've submitted a bug for that package in RedHat Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1408969 and that's I'm afraid all I can help with here as it's outside of Tomboy.

Debian and Ubuntu, for instance, have backported those commits (https://bugs.launchpad.net/ubuntu/+source/dbus-sharp/+bug/816950) and in general have kept the package updated, but in Fedora for whatever reason that seems to have fallen through the cracks (even though my usual experience with fix backporting/package updates is typically the opposite between those distros).
